include/libswiftnav/signal.h

```diff
@@ -19,15 +20,35 @@
 #define NUM_SATS_SBAS 19
 #define NUM_SATS (NUM_SATS_GPS + NUM_SATS_SBAS)
 
+#define CODES_ON_GPS_SAT 2
+#define CODES_ON_SBAS_SAT 1
+
+#define NUM_CODES_GPS (NUM_SATS_GPS * CODES_ON_GPS_SAT)
+#define NUM_CODES_SBAS (NUM_SATS_SBAS * CODES_ON_SBAS_SAT)
+#define NUM_CODES (NUM_CODES_GPS + NUM_CODES_SBAS)
+
+#define GPS_FIRST_PRN 1
+#define SBAS_FIRST_PRN 120
+
 enum constellation {
+  CONSTELLATION_INVALID = -1,
   CONSTELLATION_GPS,
   CONSTELLATION_SBAS,
   CONSTELLATION_COUNT,
 };
 
-enum band {
-  BAND_L1,
-  BAND_COUNT,
+enum code {
+  CODE_INVALID = -1,
+  CODE_GPS_L1CA,
+  CODE_GPS_L2CL,
+  CODE_GPS_L2CM,
+  CODE_SBAS_L1CA,
+  CODE_COUNT,
+};
+
+enum indexing_type {
```

Comment:
I think we want
INDEX_GLOBAL [0, NUM_SIGNALS)
INDEX_CONSTELLATION [0, NUM_SIGNALS_<constellation>)
INDEX_CODE [0, NUM_SIGNALS_<code>)